pydbus/strong_typing.py

```diff
@@ -34,3 +35,10 @@ def decorate(func):
 		verify_arguments(func)
 		return func
 	return decorate
+
+
+class TypedSignal(signal):
+
+	def __init__(self, arguments):
```

Comment:
You last suggestion is fascinating! In theory you could define signals as methods (irrespective of this patch), and then you could either use the Python 3 syntax or `@typed_method` to specify the parameter types.

So in theory `@typed_signal` doesn't need to be implemented at all. I'll try to look at that feature because then it doesn't need to introduce something that isn't necessary actually.